### DISCUSSION TITLE
Why is Zephyr parsing my VERSION file?

### DISCUSSION THREAD

**Discussion by TomzBench** [zephyrproject-rtos/zephyr#61054] [General] [answered]
Zephyr has very nice infrastructure for managing VERSION information. But I have an application that does it's own thing with a VERSION file. When I compile my zephyr app. Zephyr is assuming my applications VERSION file is conforming to it's expectations.

```cmake
# zephyr/cmake/modules/version.cmake
  set(VERSION_FILE ${ZEPHYR_BASE}/VERSION ${APPLICATION_SOURCE_DIR}/VERSION)
```

NOTE how it is parsing a VERSION file inside ${APPLICATION_SOURCE_DIR}  ...this is pretty aggressively opinionated i think. This kind of feature should be opt in...  Why can't zephyr apps have their own version management schemes?

**Comment by nordicjm** [comment]
There were 2 proposals submitted for an application version scheme, one was for an entirely driven Kconfig method: https://github.com/zephyrproject-rtos/zephyr/pull/55814 and the other uses a VERSION file: https://github.com/zephyrproject-rtos/zephyr/pull/56141

People were prompted lots of times to give feedback i.e. at the architecture working group meeting, no-one did, so since both PR creator opposed the other's system in favour of their own, the Kconfig one was dropped and VERSION one was implemented.

**Reply by TomzBench** [reply]
I see.  Is there docs on how to use it? im not seeing APP_VERSION_STRING defined in my app

**Reply by nordicjm** [reply, marked as answer]
Awaiting review on https://github.com/zephyrproject-rtos/zephyr/pull/61068

**Reply by TomzBench** [reply]
Great thanks,  and sorry for redundant posting on the matter in the PR
